api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/ListOfPatientsCurrentlyOnArtWithoutTbScreeningCohortQueries.java

```diff
@@ -0,0 +1,580 @@
+
+    return sqlPatientDataDefinition;
+  }
+
+  /**
+   * <b>Technical Specs</b>
+   *
+   * <blockquote>
+   * <p>The most recent Date of the Drug Pick Up registered on Ficha Recepção Levantou ARV by report end date.
+   *
+   * </blockquote>
+   *
+   * @return {@link DataDefinition}
+   */
+  public DataDefinition getMostRecentDrugPickupDateOnRecepcaoLevantouArv() {
+
+    SqlPatientDataDefinition sqlPatientDataDefinition = new SqlPatientDataDefinition();
+    sqlPatientDataDefinition.setName("Most Recent Drug Pick-Up on Recepção Levantou ARVs ");
+    sqlPatientDataDefinition.addParameter(new Parameter("location", "Location", Location.class));
+    sqlPatientDataDefinition.addParameter(new Parameter("endDate", "End Date", Location.class));
+
+    Map<String, Integer> valuesMap = new HashMap<>();
+    valuesMap.put("23866", hivMetadata.getArtDatePickupMasterCard().getConceptId());
+    valuesMap.put("52", hivMetadata.getMasterCardDrugPickupEncounterType().getEncounterTypeId());
+
+    String query =
+        "SELECT p.patient_id, "
+            + "       MAX(o.value_datetime) encounter_date "
+            + "FROM   patient p "
+            + "       INNER JOIN encounter e "
+            + "               ON e.patient_id = p.patient_id "
+            + "       INNER JOIN obs o "
+            + "               ON o.encounter_id = e.encounter_id "
+            + "WHERE  e.encounter_type = ${52} "
+            + "       AND e.location_id = :location "
+            + "       AND e.encounter_datetime <= :endDate "
+            + "       AND o.concept_id = ${23866} "
+            + "       AND p.voided = 0 "
+            + "       AND e.voided = 0 "
+            + "GROUP  BY p.patient_id";
+
+    StringSubstitutor sb = new StringSubstitutor(valuesMap);
+    sqlPatientDataDefinition.setQuery(sb.replace(query));
+
+    return sqlPatientDataDefinition;
+  }
+
+  /**
+   * <b>Technical Specs</b>
+   *
+   * <blockquote>
+   * <p>Last Drug Pick up Date registered on Ficha Recepção Levantou ARV by report end date (value of Column O) + 30 days
+   *
+   * </blockquote>
+   *
+   * @return {@link DataDefinition}
+   */
+  public DataDefinition getNextScheduledDrugPickupDateOnRecepcaoLevantouArv() {
+
+    SqlPatientDataDefinition sqlPatientDataDefinition = new SqlPatientDataDefinition();
+    sqlPatientDataDefinition.setName("Most Recent Drug Pick-Up on Recepção Levantou ARVs ");
+    sqlPatientDataDefinition.addParameter(new Parameter("location", "Location", Location.class));
+    sqlPatientDataDefinition.addParameter(new Parameter("endDate", "End Date", Location.class));
+
+    Map<String, Integer> valuesMap = new HashMap<>();
+    valuesMap.put("23866", hivMetadata.getArtDatePickupMasterCard().getConceptId());
+    valuesMap.put("52", hivMetadata.getMasterCardDrugPickupEncounterType().getEncounterTypeId());
+
+    String query =
+        "SELECT p.patient_id, "
+            + "       MAX(DATE_ADD(o.value_datetime, INTERVAL 30 MONTH)) encounter_date "
```

Comment:
Looks like this interval range need to be in days, NOT months

Reply:
Yes. Fixed